traversal/procedure/GraphProcedure.java

```diff
@@ -181,39 +184,39 @@ public Builder() {
             this.vertices = new HashMap<>();
         }
 
-        // TODO: remove when the query planner orders vertices natively
-        public GraphProcedure orderAndbuild() {
-            return new GraphProcedure(orderedVertices());
-        }
-
         public GraphProcedure build() {
-            return new GraphProcedure(vertices.values().stream().sorted(Comparator.comparing(ProcedureVertex::order))
+            return new GraphProcedure(vertices.values().stream().sorted(comparing(ProcedureVertex::order))
                     .toArray(ProcedureVertex[]::new));
         }
 
-        /**
-         * TODO: this is a compatibility layer to translate edge-ordered procedures into vertex-ordered procedures
-         * we need to remove it once the query planner natively orders vertices instead of edges
-         */
-        private ProcedureVertex<?, ?>[] orderedVertices() {
-            ProcedureVertex<?, ?>[] orderedVertices = new ProcedureVertex[vertices.size()];
-            List<ProcedureVertex<?, ?>> vertexList = iterate(vertices.values()).toList();
-            vertexList.sort(Comparator.comparing(v -> v.lastInEdge() == null ? 0 : v.lastInEdge().order()));
-            for (int i = 0; i < vertexList.size(); i++) {
-                orderedVertices[i] = vertexList.get(i);
-                orderedVertices[i].setOrder(i);
+        private void streamlineVertices() {
+            Stack<ProcedureVertex<?, ?>> open = new Stack<>();
+            vertices.values().stream().filter(ProcedureVertex::isStartingVertex).forEach(open::add);
+            Set<ProcedureVertex<?, ?>> closed = new HashSet<>();
+
+            int vertexOrder = 0;
+            while (!open.empty()) {
+                ProcedureVertex<?, ?> vertex = open.pop();
+                vertex.setOrder(vertexOrder++);
```

Comment:
let's aim to have this method deleted and make `ProcedureVertex` more stateless

Reply:
`ProcedureVertex::order()` is used in a ton of places in `GraphIterator`, I think that change might be better suited for its own PR.